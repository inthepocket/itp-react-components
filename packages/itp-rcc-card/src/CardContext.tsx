import { createContext } from 'react';
import CardStylesInterface from './CardStylesInterface';

interface CardContextInterface {
  prefixCss?: string | null;
  size: 'small' | 'default' | 'large';
  styles?: CardStylesInterface | null;
}

/**
 * Card Context
 */
const CardContext =
  createContext<CardContextInterface>({ prefixCss: 'card', size: 'default', styles: null });

export default CardContext;
