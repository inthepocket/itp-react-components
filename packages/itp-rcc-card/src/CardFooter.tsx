import * as React from 'react';
import capitalize from 'capitalize';
import classnames from 'classnames';
import CardContext from './CardContext';
import CardChildPropsInterface from './CardChildPropsInterface';

const { Consumer } = CardContext;

/**
 * Card Footer
 */
const CardFooter = ({ children }: CardChildPropsInterface) => {
  return (
    <Consumer>
      {(cardContext) => {
        const { prefixCss, size, styles } = cardContext;
        const containerClassName = styles ? styles.footer : `${prefixCss}-footer`;
        const sizeClassName = styles ? styles[`footerSize${capitalize(size)}`] : `${prefixCss}-footer-size-${size}`;

        const className = classnames({
          [`${containerClassName}`]: containerClassName,
          [`${sizeClassName}`]: sizeClassName,
        });

        return (
          <footer className={className}>{children}</footer>
        );
      }}
    </Consumer>
  );
};

export default CardFooter;
