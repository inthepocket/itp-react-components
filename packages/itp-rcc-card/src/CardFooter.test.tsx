import * as React from 'react';
import { create } from 'react-test-renderer';
import CardFooter from './CardFooter';
import CardContext from './CardContext';

const { Provider } = CardContext;
const cssModuleMock = {
  body: 'cssModulesBody',
  footer: 'cssModulesFooter',
  footerSizeSmall: 'cssModulesFooterSizeSmall',
  footerSizeLarge: 'cssModulesFooterSizeLarge',
  card: 'cssModulesCard',
};

test('CardFooter: default', () => {
  const tree = create((
    <CardFooter>
      <p>Cras sed ante. Phasellus in massa. Curabitur dolor eros, gravida et, hendrerit ac, cursus non, massa. Aliquam lorem.</p>
    </CardFooter>
  ))
  .toJSON();

  expect(tree).toMatchSnapshot();
});

test('CardFooter: size: small', () => {
  const tree = create((
    <Provider value={{ prefixCss: 'card', size: 'small' }}>
      <CardFooter>
        <p>Cras sed ante. Phasellus in massa. Curabitur dolor eros, gravida et, hendrerit ac, cursus non, massa. Aliquam lorem.</p>
      </CardFooter>
    </Provider>
  ))
  .toJSON();

  expect(tree).toMatchSnapshot();
});

test('CardFooter: size: large', () => {
  const tree = create((
    <Provider value={{ prefixCss: 'card', size: 'large' }}>
      <CardFooter>
        <p>Cras sed ante. Phasellus in massa. Curabitur dolor eros, gravida et, hendrerit ac, cursus non, massa. Aliquam lorem.</p>
      </CardFooter>
    </Provider>
  ))
  .toJSON();

  expect(tree).toMatchSnapshot();
});

test('CardFooter: css modules: default', () => {
  const tree = create((
    <Provider value={{ prefixCss: 'card', size: 'default', styles: cssModuleMock }}>
      <CardFooter>
        <p>Cras sed ante. Phasellus in massa. Curabitur dolor eros, gravida et, hendrerit ac, cursus non, massa. Aliquam lorem.</p>
      </CardFooter>
    </Provider>
  ))
  .toJSON();

  expect(tree).toMatchSnapshot();
});

test('CardFooter: css modules: size: small', () => {
  const tree = create((
    <Provider value={{ prefixCss: 'card', size: 'small', styles: cssModuleMock }}>
      <CardFooter>
        <p>Cras sed ante. Phasellus in massa. Curabitur dolor eros, gravida et, hendrerit ac, cursus non, massa. Aliquam lorem.</p>
      </CardFooter>
    </Provider>
  ))
  .toJSON();

  expect(tree).toMatchSnapshot();
});

test('CardFooter: css modules: size: large', () => {
  const tree = create((
    <Provider value={{ prefixCss: 'card', size: 'large', styles: cssModuleMock }}>
      <CardFooter>
        <p>Cras sed ante. Phasellus in massa. Curabitur dolor eros, gravida et, hendrerit ac, cursus non, massa. Aliquam lorem.</p>
      </CardFooter>
    </Provider>
  ))
  .toJSON();

  expect(tree).toMatchSnapshot();
});