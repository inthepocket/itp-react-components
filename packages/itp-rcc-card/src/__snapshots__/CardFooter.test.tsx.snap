// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardFooter: css modules: default 1`] = `
<footer
  className="cssModulesFooter"
>
  <p>
    Cras sed ante. Phasellus in massa. Curabitur dolor eros, gravida et, hendrerit ac, cursus non, massa. Aliquam lorem.
  </p>
</footer>
`;

exports[`CardFooter: css modules: size: large 1`] = `
<footer
  className="cssModulesFooter cssModulesFooterSizeLarge"
>
  <p>
    Cras sed ante. Phasellus in massa. Curabitur dolor eros, gravida et, hendrerit ac, cursus non, massa. Aliquam lorem.
  </p>
</footer>
`;

exports[`CardFooter: css modules: size: small 1`] = `
<footer
  className="cssModulesFooter cssModulesFooterSizeSmall"
>
  <p>
    Cras sed ante. Phasellus in massa. Curabitur dolor eros, gravida et, hendrerit ac, cursus non, massa. Aliquam lorem.
  </p>
</footer>
`;

exports[`CardFooter: default 1`] = `
<footer
  className="card-footer card-footer-size-default"
>
  <p>
    Cras sed ante. Phasellus in massa. Curabitur dolor eros, gravida et, hendrerit ac, cursus non, massa. Aliquam lorem.
  </p>
</footer>
`;

exports[`CardFooter: size: large 1`] = `
<footer
  className="card-footer card-footer-size-large"
>
  <p>
    Cras sed ante. Phasellus in massa. Curabitur dolor eros, gravida et, hendrerit ac, cursus non, massa. Aliquam lorem.
  </p>
</footer>
`;

exports[`CardFooter: size: small 1`] = `
<footer
  className="card-footer card-footer-size-small"
>
  <p>
    Cras sed ante. Phasellus in massa. Curabitur dolor eros, gravida et, hendrerit ac, cursus non, massa. Aliquam lorem.
  </p>
</footer>
`;
